samples/extensions/dynamic_rendering_local_read/dynamic_rendering_local_read.cpp

```diff
@@ -0,0 +1,1052 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "dynamic_rendering_local_read.h"
+#include "scene_graph/components/mesh.h"
+#include "scene_graph/components/pbr_material.h"
+#include "scene_graph/components/sub_mesh.h"
+
+DynamicRenderingLocalRead::DynamicRenderingLocalRead()
+{
+	title = "Dynamic Rendering local read";
+
+	camera.type = vkb::CameraType::FirstPerson;
+	camera.set_position({0.0f, 0.5f, 0.0f});
+	camera.set_position({-3.2f, 1.0f, 5.9f});
+	camera.set_rotation({0.5f, 210.05f, 0.0f});
+	camera.set_perspective(60.f, static_cast<float>(width) / static_cast<float>(height), 256.f, 0.1f);
+
+#if defined(USE_DYNAMIC_RENDERING)
+	set_api_version(VK_API_VERSION_1_2);
+	add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+	add_device_extension(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
+	add_device_extension(VK_KHR_DYNAMIC_RENDERING_LOCAL_READ_EXTENSION_NAME);
+	// To simplify barrier setup used for dynamic rendering, we use sync2
+	add_device_extension(VK_KHR_SYNCHRONIZATION_2_EXTENSION_NAME);
+
+	// Dynamic rendering doesn't use render passes
+	// To make sure that framework related classes like the user interface are aware of this, we explicitly st the base class' renderpass to a null handle
+	render_pass = VK_NULL_HANDLE;
+
+	LOGI("Using dynamic rendering with local read");
+#endif
+}
+
+DynamicRenderingLocalRead::~DynamicRenderingLocalRead()
+{
+	if (device)
+	{
+		for (Pass pass : {scene_opaque_pass, scene_transparent_pass, composition_pass})
+		{
+			vkDestroyPipeline(get_device().get_handle(), pass.pipeline, nullptr);
+			vkDestroyPipelineLayout(get_device().get_handle(), pass.pipeline_layout, nullptr);
+			vkDestroyDescriptorSetLayout(get_device().get_handle(), pass.descriptor_set_layout, nullptr);
+		}
+		for (FrameBufferAttachment attachment : {attachments.albedo, attachments.normal, attachments.positionDepth})
+		{
+			destroy_attachment(attachment);
+		}
+		vkDestroySampler(get_device().get_handle(), textures.transparent_glass.sampler, nullptr);
+	}
+}
+
+void DynamicRenderingLocalRead::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = true;
+	}
+#if defined(USE_DYNAMIC_RENDERING)
+	auto &requested_dynamic_rendering_features            = gpu.request_extension_features<VkPhysicalDeviceDynamicRenderingFeaturesKHR>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DYNAMIC_RENDERING_FEATURES_KHR);
```

Comment:
Does that make sense?
We get the `VkPhysicalDeviceDynamicRenderingFeaturesKHR` here, with all supported flags set (well, it's just one). No matter what, we set the `dynamicRendering` flag.
Shouldn't we make it similar to how we handle the `samplerAnisotropy` flag above?

Reply:
This is not specific to this sample, though.
Might be worth to file an issue on it...

Reply:
See #938